app/helpers/react_on_rails_helper.rb

```diff
@@ -2,132 +2,196 @@
 
 module ReactOnRailsHelper
 
-  # component_name: React component name
+  # react_component_name: can be a React component, created using a ES6 class, or
+  #   React.createClass, or a
+  #     `generator function` that returns a React component AND this function must have the
+  #       property "generator" set to true
+  #       using ES6
+  #          let MyReactComponentApp = (props) => <MyReactComponent {...props}/>;
+  #          MyReactComponentApp.generator = true;
+  #       or using ES5
+  #          var MyReactComponentApp = function(props) { return <YourReactComponent {...props}/>; }
+  #          MyReactComponentApp.generator = true;
+  #    Exposing the react_component_name is necessary to both a plain ReactComponent as well as
+  #      a generator:
+  #    For client rendering, expose the react_component_name on window:
+  #      window.MyReactComponentApp = MyReactComponentApp;
+  #    For server rendering, export the react_component_name on global:
+  #      global.MyReactComponentApp = MyReactComponentApp;
+  #    See spec/dummy/client/app/startup/serverGlobals.jsx and
+  #      spec/dummy/client/app/startup/ClientApp.jsx for examples of this
   # props: Ruby Hash which contains the properties to pass to the react object
   #
-  # Naming Conventions:
-  # Suppose your component is named "App" (can be anything)
-  # 1. Inside app/startup/ServerApp.jsx, setup the component for server rendering. This is used by
-  #    the global exports in step 2
-  # 2. Component_name is changed to CamelizedUpper for the exposed component,
-  #    so we have App as the exposed global (see app/startup/serverGlobals.jsx)
-  # 3. Inside app/startup/ClientApp.jsx, you want to export a function that generates
-  #    your component, and when you generate the component, you must use this naming convention
-  #    to get the data.
-  #    App(__appData__)
-  #    That way it gets the data you pass from the Rails helper.
-  #
   #  options:
-  #    prerender: <true/false> defaults to true
-  #    trace: <true/false> defaults to false, set to true to print additional info
+  #    prerender: <true/false> (configurable default), set to false when debugging!
+  #    trace: <true/false>  (configurable default), set to true to print additional info
   def react_component(component_name, props = {}, options = {})
```

Comment:
Ah, interesting! I hardly ever have method calls for the options. I can see how this could work. @josiasds Please give this a try.